v0/destinations/hs/transform.js

```diff
@@ -292,22 +299,31 @@ function batchEvents(destEvents) {
     let batchEventResponse = defaultBatchRequestConfig();
 
     chunk.forEach(ev => {
-      const { email, updatedProperties } = getEmailAndUpdatedProps(
-        ev.message.body.JSON.properties
-      );
-      ev.message.body.JSON.properties = updatedProperties;
-      identifyResponseList.push({
-        email,
-        properties: ev.message.body.JSON.properties
-      });
-      metadata.push(ev.metadata);
+      const { properties } = ev.message.body.JSON
+      if (properties && !(properties instanceof Array)) {
+        identifyResponseList.push({...ev.message.body.JSON})
```

Comment:
Pushing both `custom object` transformed payload and `create or update contacts` transformed payload into single `arrayChunksIdentify = []` will result in incorrect batching up items.

The result would be a mixture of both the endpoint's payload batched up together also as per the current logic the final batched payload's endpoint will be dependent on the last chuck which will be received. 

say the last chunk was for `create/update contacts`, therefore, the final batched payload will go to its batch endpoint and if the last chunk is of `custom objects` the final mixture of the batched payloads will go to the batch custom objects endpoint.

Reply:
Chunks need to be created based on the endpoint at Line: 277

Reply:
For any object the end points are same... 
https://api.hubspot.com/crm/v3/objects/${objectType}. -- non batch
${ev.message.endpoint}/batch/create` -- batch

For these the body is different from how other sources will be handled. Here the json body will be an object whereas for all other cases of identify it will be array thus handled this way.

Reply:
Resolving this as discussed.